File: bungee/src/main/java/net/impactdev/gts/bungee/messaging/interpreters/BungeeAdminInterpreters.java
package net.impactdev.gts.bungee.messaging.interpreters;

import net.impactdev.gts.api.messaging.IncomingMessageConsumer;
import net.impactdev.gts.common.messaging.interpreters.Interpreter;
import net.impactdev.gts.common.messaging.messages.admin.ForceDeleteMessageImpl;
import net.impactdev.gts.common.plugin.GTSPlugin;
import net.impactdev.gts.common.utils.exceptions.ExceptionWriter;

public class BungeeAdminInterpreters implements Interpreter {

    @Override
    public void register(GTSPlugin plugin) {
        this.getDecoders(plugin);
        this.getInterpreters(plugin);
    }

    @Override
    public void getDecoders(GTSPlugin plugin) {
        plugin.messagingService().registerDecoder(
                ForceDeleteMessageImpl.ForceDeleteRequest.TYPE, ForceDeleteMessageImpl.ForceDeleteRequest::decode
        );
    }

    @Override
    public void getInterpreters(GTSPlugin plugin) {
        final IncomingMessageConsumer consumer = plugin.messagingService().getMessenger().getMessageConsumer();

        consumer.registerInternalConsumer(
                ForceDeleteMessageImpl.ForceDeleteRequest.class, request -> {
                    request.respond().thenAccept(response -> {
                        GTSPlugin.instance().messagingService().getMessenger().sendOutgoingMessage(response);
                        GTSPlugin.instance().logger().info("Response sent");
                    })
                    .exceptionally(e -> {
                        ExceptionWriter.write(e);
                        return null;
                    });
                }
        );
    }
}

File: bungee/src/main/kotlin/net/impactdev/gts/bungee/messaging/interpreters/BungeeAdminInterpreters.kt
package net.impactdev.gts.bungee.messaging.interpreters

import com.google.gson.JsonElement
import net.impactdev.gts.api.messaging.message.MessageConsumer
import net.impactdev.gts.api.messaging.message.type.admin.ForceDeleteMessage
import net.impactdev.gts.common.messaging.interpreters.Interpreter
import net.impactdev.gts.common.messaging.messages.admin.ForceDeleteMessageImpl.ForceDeleteRequest
import net.impactdev.gts.common.plugin.GTSPlugin
import net.impactdev.gts.common.utils.exceptions.ExceptionWriter
import java.util.*
import java.util.function.Consumer

class BungeeAdminInterpreters : Interpreter {
    override fun register(plugin: GTSPlugin) {
        getDecoders(plugin)
        getInterpreters(plugin)
    }

    override fun getDecoders(plugin: GTSPlugin) {
        plugin.messagingService.registerDecoder(
            ForceDeleteRequest.TYPE
        ) { content: JsonElement?, id: UUID? -> ForceDeleteRequest.decode(content, id) }
    }

    override fun getInterpreters(plugin: GTSPlugin) {
        val consumer = plugin.messagingService.messenger.messageConsumer
        consumer!!.registerInternalConsumer(
            ForceDeleteRequest::class.java, MessageConsumer<ForceDeleteRequest> { request: ForceDeleteRequest ->
                request.respond()!!
                    .thenAccept(
                        Consumer<ForceDeleteMessage.Response> { response: ForceDeleteMessage.Response? ->
                            GTSPlugin.instance.messagingService.messenger.sendOutgoingMessage(
                                response!!
                            )
                            GTSPlugin.instance.pluginLogger.info("Response sent")
                        })
                    .exceptionally { e: Throwable? ->
                        ExceptionWriter.write(e)
                        null
                    }
            }
        )
    }
}